src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +816,49 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        private static void DetectAndMarkAmbiguousCentralTransitiveDependencies<TItem>(this GraphNode<TItem> root, Tracker<TItem> tracker)
+        {
+            // if a central transitive node has all parents disputed or ambiguous mark it and its children ambiguous
+            root.InnerNodes
+                .Where(x => x.Item.IsCentralTransitive && x.Disposition == Disposition.Acceptable)
+                .ForEach(n =>
+                {
+                    if (!n.ParentNodes.Any(p => !(tracker.IsDisputed(p.Item) || tracker.IsAmbiguous(p.Item))))
+                    {
+                        tracker.MarkAmbiguous(n.Item);
+                        n.ForEach(x => tracker.MarkAmbiguous(x.Item));
```

Comment:
Why do you need to do this explicit walk of child nodes here? Can you do this in a single pass without the nested ForEach by moving the predicate?

Reply:
It will be a single pass over the central transitive nodes. I did few updates, i am not sure if i addressed your concerns though.